hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -110,6 +112,9 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         if (op.hasAdminKey()) {
             context.requireKey(op.adminKeyOrThrow());
         }
+        if (op.hasFeeScheduleKey()) {
+            validateTruePreCheck(isValid(op.feeScheduleKey()), INVALID_CUSTOM_FEE_SCHEDULE_KEY);
```

Comment:
@Neeharika-Sompalli, @mhess-swl: Is this the right place for such a check? Shouldn't this be checked in handle?

Reply:
btw this was my thought as well, but moving it to handle and adjusting the test, breaks it in the mono

Reply:
Actually digging into it some more, it looks like `isValid()` does not rely on state, and therefore this check can be part of the `pureCheck()` method IMO.

Reply:
I can move it there, seems logical to me also, let's see what @Neeharika-Sompalli and/or @mhess-swl also think ?

Reply:
@georgi-l95  yes it doesn't depend on state. It can be moved to pureCheck